### PR TITLE
Add site title component

### LINKED ISSUE #69: Add profile link component
The bottom portion of the navigation container includes a link to the user's profile.  We'll need to add the user avatar data and pull the current user info out of the menu items data store.

Screen Shot 2020-08-28 at 11 38 59 AM

### LINKED ISSUE #70: Add site link component
The top portion of the navigation container contains a site link component that houses a link to the frontend of the site and the site title.

Screen Shot 2020-08-28 at 11 40 07 AM

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #70 

Adds the site title component.

A follow will probably be needed:
* Site title/url should probably be migrated to `@wordpress/core-data`.

### Screenshots
Screen Shot 2020-09-04 at 6 13 59 PM
Screen Shot 2020-09-04 at 6 13 52 PM

### Testing
1. Open any page with the wc nav.
1. Make sure the nav is shown with your site title.
1. Click on the site title to make sure it navigates to your site's frontend.
1. Click on the site icon and make sure it folds the menu.

/cc @jameskoster This probably needs a design review.  The folded state in particular has a couple items that may need updating:
* The height of the header is 60px but the WCA header as seen above is 56px so they don't quite line up.
* Currently this folds with 60px remaining on the left (so there will always be a gap).  Is this okay or should it allow content to be flush left?  (We'll need to modify the WCA header slightly to accommodate if so.

### CODE REVIEW

File: client/components/navigation-header/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		document.body.classList.toggle( 'is-folded' );
+	};
+
+	let buttonIcon = <Icon size="36px" icon={ wordpress } />;
+
+	if ( siteIconUrl ) {
+		buttonIcon = (
+			<img
+				alt={ __( 'Site Icon' ) }
+				className="edit-post-fullscreen-mode-close_site-icon"
```
Comment (jeffstieler) [inline client/components/navigation-header/index.js:31]: Looks like this is just a hold over from `FullscreenModeClose`? Do we need the class at all? If so, should it be renamed to be more relevant to the usage here?

File: client/components/navigation-header/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import { compose } from '@wordpress/compose';
+import { Icon, wordpress } from '@wordpress/icons';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { NAVIGATION_STORE_NAME } from '../../store';
+
+const NavigationContainer = ( {
```
Comment (jeffstieler) [inline client/components/navigation-header/index.js:15]: This `NavigationContainer` component is in a `navigation-header` directory, but imported as `SiteTitle`. What do you think about picking one name for consistency/readability?

_(We totally do this in other repos, I'd like us to break the habit 😄 )_

File: client/components/navigation-header/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+				as="span"
+			>
+				{ siteTitle }
+			</Button>
+		</div>
+	);
+};
+
+export default compose(
+	withSelect( ( select ) => {
```
Comment (jeffstieler) [inline client/components/navigation-header/index.js:59]: What do you think about using the `useSelect()` hook instead?

File: webpack.config.js [hunk 4]
```diff
@@ -17,4 +18,18 @@ module.exports = {
 			react: path.resolve( './node_modules/react' ),
 		},
 	},
+	plugins: [
+		...defaultConfig.plugins.filter(
+			( plugin ) =>
+				plugin.constructor.name !== 'DependencyExtractionWebpackPlugin'
+		),
+		new DependencyExtractionWebpackPlugin( {
```
Comment (jeffstieler) [inline webpack.config.js:26]: Why was this needed? Should we add a comment explaining it?

**Review by jeffstieler — COMMENTED** [review]
This tested well for me. Just leaving some comments for now since you'd like a design review from @jameskoster.

File: client/components/navigation-header/index.js (hunk 2 above)
Comment (joshuatf) [reply client/components/navigation-header/index.js]: You're right.  Guess I was playing around with the names and forgot to make them consistent.  Updated.

File: webpack.config.js (hunk 4 above)
Comment (joshuatf) [reply webpack.config.js:28]: Added a comment here.  Needed to prevent `@wordpress/components` from being loaded via externals.

File: client/components/navigation-header/index.js (hunk 1 above)
Comment (joshuatf) [reply client/components/navigation-header/index.js]: This was lazy copy pasta.  Not needed and removed.

**Comment by joshuatf** [comment]
Thanks for reviewing this, @jeffstieler.  Handled feedback and gave this a rebase.  The `Navigation` component has undergone extensive changes in GB so it may need to be commented out and just review `<NavigationHeader/>` in this PR.

https://github.com/woocommerce/navigation/pull/83 will address updates needed to get the navigation menus working again.

**Comment by psealock** [comment]
Locally, I rebased this against `add/composibiliity-integration` from https://github.com/woocommerce/navigation/pull/83 and ran into some issues. 

Screen Shot 2020-09-15 at 10 17 36 AM

Look familiar at all?

**Comment by joshuatf** [comment]
@psealock When I rebase against #83 and when I have `master` checked out in GB and linked, this works as expected for me.

Screen Shot 2020-09-15 at 4 42 29 PM

File: client/components/navigation-container/style.scss [hunk 5]
```diff
@@ -5,6 +5,7 @@
 	width: $navigation-width;
 	height: 100%;
 	box-sizing: border-box;
+	background-color: #1e1e1e;
```
Comment (psealock) [inline client/components/navigation-container/style.scss:8]: Lets add a note to remind ourselves this is `G2.darkGray.primary`, which I don't believe is exported just yet.

File: src/Loader.php [hunk 6]
```diff
@@ -72,15 +72,22 @@ public static function register_navigation_script() {
...
 
 		wp_register_script(
 			'woocommerce-navigation',
 			$script_url,
-			$script_asset['dependencies'],
+			array(
+				'wp-core-data',
+				'wp-date',
+				'wp-keycodes',
+			),
```
Comment (psealock) [inline src/Loader.php:90]: This should be handled automatically by `@wordpress/dependency-extraction-webpack-plugin`. `$script_asset['dependencies']` should be populated with the correct strings. If its not, our usage is wrong somehow because its whole purpose is to avoid having to explicitly name things like this.

File: src/Loader.php [hunk 7]
```diff
@@ -72,15 +72,22 @@ public static function register_navigation_script() {
 		$script_asset      = file_exists( $script_asset_path )
 			? require $script_asset_path
 			: array(
-				'dependencies' => array(),
+				'dependencies' => array(
+					'wp-core-data',
+					'wp-date',
```
Comment (psealock) [inline src/Loader.php:77]: Do we need wp-date?

**Review by psealock — COMMENTED** [review]
I am having a real hard time with `npm link` on this branch. I `rm -rf`'d Gutenberg plugin and these `node_modules` to start over fresh but no luck.

I left a couple comments from looking at the code, and hopefully I can take it for a spin once #83 is merged and this is rebased.

File: src/Loader.php (hunk 6 above)
Comment (joshuatf) [reply src/Loader.php]: Had to import `@wordpress/core-data` manually to get this working since `@wordpress/dependency-extraction-webpack-plugin` automatically maps an externals starting with `@wordpress`, but appears to be working now and removed these manual overrides.

File: src/Loader.php (hunk 7 above)
Comment (joshuatf) [reply src/Loader.php]: From an original implementation prior to using `@wordpress/dependency-extraction-webpack-plugin`- removed.

File: client/components/navigation-container/style.scss (hunk 5 above)
Comment (joshuatf) [reply client/components/navigation-container/style.scss:9]: Done!

**Comment by joshuatf** [comment]
Addressed your feedback, @psealock and gave this a rebase.  Looks to be working on my end, but let me know if you have any trouble running.

**Review by psealock — APPROVED** [review]
Thanks for the rebase and fixes @joshuatf, npm is linking correctly for me now. Code looks good and testing well. I did just notice some styles in the buttons making focus state look cramped. Otherwise, 🚢 

Screen Shot 2020-09-17 at 8 41 58 AM
Screen Shot 2020-09-17 at 8 44 07 AM

**Comment by jameskoster** [comment]
> The height of the header is 60px but the WCA header as seen above is 56px so they don't quite line up.

I don't know how much extra work would be involved, but we should probably update WCA to match GB. There's a var for this [here](https://github.com/WordPress/gutenberg/blob/master/packages/base-styles/_variables.scss#L44).

> Currently this folds with 60px remaining on the left (so there will always be a gap). Is this okay or should it allow content to be flush left? (We'll need to modify the WCA header slightly to accommodate if so.

I'm not 100% sure what you mean here, sorry 😬 But one thing I think may be worth discussing is whether the nav should be collapsible in Woo context. It may be that the nav should just be permanently visible and un-collapsible in WCA?

**Comment by joshuatf** [comment]
Going to go ahead and merge this so we can take advantage of the dependency injection changes in this PR.

>I don't know how much extra work would be involved, but we should probably update WCA to match GB. There's a var for this here.

@jameskoster Great- will update on the WCA side.

>I'm not 100% sure what you mean here, sorry 😬 But one thing I think may be worth discussing is whether the nav should be collapsible in Woo context. It may be that the nav should just be permanently visible and un-collapsible in WCA?

Personally, I don't see the harm in allowing it to be collapsed if clicked, but I don't have a strong opinion on this.

Also in response to https://github.com/woocommerce/navigation/pull/74#pullrequestreview-490010000 - the current padding fits the designs to the pixel, but results in the cramped focus styles that @psealock mentioned.  Is this okay or should we update this to increase the padding for the site title (which will also increase the overall spacing between the logo and site title)?

**Comment by psealock** [comment]
>Is this okay or should we update this to increase the padding for the site title (which will also increase the overall spacing between the logo and site title)?

Can we just split the difference of the padding-right so its on both sides?

Screen Shot 2020-09-17 at 8 44 07 AM

**Comment by jameskoster** [comment]
Ah the focus styles! I would probably just reduce the left and right padding values to 0.
